### DISCUSSION TITLE
Types checking with Vue in Vite

### DISCUSSION THREAD

**Discussion by jzs11** [vuejs/language-tools#530] [Q&A] [answered]
Hello buddy,

I am seeking some advice here again 😊

I recently transferred my project from using vue-cli (webpack) to Vite. However, there is a problem that the type checking does not work as I expected. Regards to @yyx990803 - https://github.com/vitejs/vite/issues/2539#issuecomment-800291208 

If I put something wrong in the IDE like the following picture

![image](https://user-images.githubusercontent.com/2770401/134768339-5c09d19b-80b0-4720-8add-cc38db66f743.png)

The error can shows when the file open. However, the error will disappear if the file closes and Vite does not report any errors at all. If some bad code refactoring happens, I won't detect the errors during the compile time.

Any advice would be much appreciated.

Kind regards,
Jing

**Comment by johnsoncodehk** [comment, marked as answer]
See #53 **Instead of `tsc --noEmit`** section.

It is very expensive to verify all closed files, so Volar don't do it automatically.

**Reply by jzs11** [reply]
Cool, I see.  

What's your view on this issue - https://github.com/vitejs/vite/discussions/5085? 

Is it a Vite bug? Or Tsc bug.

**Reply by johnsoncodehk** [reply]
I think https://github.com/vitejs/vite/issues/2539#issuecomment-800291208 answered, vite don't check TS types problems, the second problem of https://github.com/vitejs/vite/discussions/5085 is syntax error not types error so vite report it.

**Reply by jzs11** [reply]
I just found out there is a setting in Volar you can validate all files!!! That's awesome. I wish the vite build pipeline could call it somehow!

**Reply by johnsoncodehk** [reply]
Use https://github.com/johnsoncodehk/vue-tsc
